cpp/src/io/csv/csv_common.h

```diff
@@ -35,6 +38,13 @@ enum : uint8_t {
 };
 using flags = uint8_t;
 
+struct column_builder {
+  cudf::data_type type;
+  void *data;
```

Comment:
Would be neat to use a view type here that indicates the buffer size (so the builder does not depend on other parameters in the call). Since we don't have anything like `device_vector_view` or `device_buffer_view`, my suggestion was to implement some sort of `device_span`, which would be more or less equivalent to std::span.

Reply:
Working on this over in #6242 :+1:

Reply:
fwiw, these changes are over in #6213